test/e2e/framework/autoscaling/autoscaling_utils.go

```diff
@@ -375,15 +375,18 @@ func (rc *ResourceConsumer) sendConsumeCPURequest(ctx context.Context, millicore
 		return true, nil
 	})
 
-	framework.ExpectNoError(err)
+	// consuming resources is a repeating background operation, getting cancelled is not a failure
+	if err != context.Canceled {
+		framework.ExpectNoError(err)
+	}
```

Comment:
What happens if the timeout is reached before the test is complete? Is that a test failure? As it stands now, that would be silently ignored while previously it was a failure.

Perhaps remove `ctx, cancel := context.WithTimeout` by calling `PollUntilContextTimeout` and then check `ctx.Err` here?
```
// If the test has already finished, ignore the result of PollUntilContextTimeout
// because it's going to be "context canceled". A "context canceled" caused by
// the local timeout is a test failure.
if ctx.Err() == nil {
    framework.ExpectNoError(err)
}
```

Reply:
> What happens if the timeout is reached before the test is complete?

The test should then fail by assertions in the specific test scenario, but you're right it's better to fail here to make debugging easier.

I like the change, thanks! Switched the if and reworded the comment in a way that is more understandable to me, please have a look :)